### ISSUE TITLE
requires gstreamer-0.10, but osdeps provide gstreamer-1.0 on ubuntu 16.04

### ISSUE THREAD

**Issue by pierrewillenbrockdfki** [rock-drivers/drivers-camera_usb#5]
Api changed between gstreamer-0.10 and gstreamer-1.0, so either it needs to be supported explicitly in here, or the manifest needs to be changed so it depends on an osdep providing gstreamer-0.10 only. 

Note that the other package depending on gstreamer is aravis, and that can handle both gstreamer versions.

**Comment by planthaber** [comment]
This is getting more severe, as Ubuntu 18.04 doesn't have a gstreamer 0.10 package at all

**Comment by maltewi** [comment]
Hi @pierrewillenbrockdfki , @planthaber any news on this issue?

A very brief look at driver/camera_usb and drivers/orogen/camera_usb tells me:

1. porting is not simply exchanging the library there are significatn api changes in between gst0.1 and gst1.0
2. gst not really used. orogen/camera_usb uses v4l. Can you confirm this?

**Comment by planthaber** [comment]
Hi,
i have no idea, but if you need a camera reading task quick, you could also give the video_streamer_vlc::Capturer Task a try.

https://github.com/rock-drivers/drivers-orogen-video_streamer_vlc

**Comment by maltewi** [comment]
It's not so much about being quick. More about how to solve this issue.

One option of cause would be to abandon this package, if `drivers-orogen-video_streamer_vlc` is a valid replacement. So, what's your opinion whether `drivers-orogen-video_streamer_vlc` could fully replace `drivers/orogen/camera_usb`?

**Comment by planthaber** [comment]
I guess rather no, because it adds an extra interface layer (vlc) on to of V4L.

I'd rather go and try to use with using opencv directly for capturing and jpeg conversion in this driver trying to replace gst.

**Comment by planthaber** [comment]
I just tried removing gst from the lib and was able to run the testmain and gram images using v4l
Can someone test it in a rock deployment?

https://github.com/rock-drivers/drivers-camera_usb/tree/no_gst_dirty/src

If it works, we can make a proper removal of the gst mode

**Comment by pierrewillenbrockdfki** [comment]
Imho whoever needs to use gst can always revert the deletion and fix the driver. Or stay on an old revision.

**Comment by planthaber** [comment]
True, but we should properly remove it from the lib then, not like i did by leaving the files an putting comments everywhere. By branch was made just to make sure it still works before doing that effort

**Comment by maltewi** [comment]
Ok, I'll have a look then to somehow get it running again. Should not take too long.. I hope

**Comment by maltewi** [comment]
Driver and orogen-component are working again on current Ubuntu.
It turned out that porting to gstreamer1.0 is less effort than removing gstreamer completely as was discusses as an alternative option above.

So: Update to use gstreamer 1.0. The patch is ready for review and imo merge at #6. Will someone of you have a look?
